docs/developer/pandas/partition_api.rst

```diff
@@ -0,0 +1,70 @@
+Pandas Partition API in Modin
+=============================
+
+If you are working with Modin DataFrame, you can unwrap its remote partitions for your needs
+(pass them to another function that will be processed on a concrete node of the cluster,
+for instance) and get IPs of the remote partitions as well. In that case you can pass the partitions
+having needed IPs to your function. It can help with minimazing of data movement between nodes. However,
+it is worth noting that for Modin on ``Ray`` engine with ``pandas`` backend IPs of the remote partitions may not match
```

Comment:
Separate discussion on Ray into separate section

Reply:
The PR is going to be divided into two. The first is for refactor, the second is for documentation. This comment will be applied in the second PR.